### PR TITLE
Optimize dataframe writer (small change)

### PR DESCRIPTION
**Pull request by Skylion007**
## Description of changes:

* Calling this method is WAY faster than calling count() on a dataframe, and will work ever on some common errors without lazy evaluation (ie missing files earlier in the pipeline)
* You should use `raise from` for raising exceptions so that the original exception information is not lost.

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [X] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [X] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [X] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [X] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [X] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [X] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#submitting-a-contribution))
- [X] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/base/converters/dataframe_to_mds.py [hunk 1]
```diff
@@ -237,16 +237,16 @@ def write_mds(iterator: Iterable):
                 for sample in records:
                     try:
                         mds_writer.write(sample)
-                    except:
-                        raise RuntimeError(f'failed to write sample: {sample}')
+                    except Exception as ex:
+                        raise RuntimeError(f'failed to write sample: {sample}') from ex
                         count += 1
```
Comment (Skylion007) [inline streaming/base/converters/dataframe_to_mds.py:242]: This is kinda useless anyway because this code is unreachable...

Comment (XiaohanZhangCMU) [reply streaming/base/converters/dataframe_to_mds.py:242]: Thanks. I thought I deleted it. It must slipped from the ~300 conversations! Can you remove counting from the function in this PR as well?

**Review by XiaohanZhangCMU — CHANGES_REQUESTED** [review]

File: streaming/base/converters/dataframe_to_mds.py (hunk 1 above)
Comment (XiaohanZhangCMU) [reply streaming/base/converters/dataframe_to_mds.py:242]: @Skylion007 have you found any better ways to surface the workers' debug output to SparkUI?

Comment (Skylion007) [reply streaming/base/converters/dataframe_to_mds.py:242]: The raise from is definitely a better way.

**Review by XiaohanZhangCMU — APPROVED** [review]

**Pull request merged**